harness/determined/common/experimental/experiment.py

```diff
@@ -52,14 +53,22 @@ def __init__(
         self._id = experiment_id
         self._session = session
 
+        try:
+            _ = self._get()
+        except errors.BadRequestException as e:
+            if isinstance(e, errors.NotFoundException):
```

Comment:
I want to make this robust to login/network/etc errors, but since `NotFoundException` subclasses `APIError` I think this catch/re-raise is needed? Maybe this is a sign that something in the exception hierarchy should be reworked?